source/common/upstream/original_dst_cluster.cc

```diff
@@ -99,12 +100,27 @@ OriginalDstCluster::LoadBalancer::requestOverrideHost(LoadBalancerContext* conte
     } else {
       ENVOY_LOG(debug, "original_dst_load_balancer: invalid override header value. {}",
                 request_override_host);
-      parent_->info()->stats().original_dst_host_invalid_.inc();
+      stats_.original_dst_host_invalid_.inc();
     }
   }
   return request_host;
 }
 
+std::unique_ptr<OriginalDstCluster::LoadBalancer::OriginalHostProvider>
+OriginalDstCluster::LoadBalancer::createFromLbConfig(
+    ClusterStats& stats,
+    const absl::optional<envoy::config::cluster::v3::Cluster::OriginalDstLbConfig>& config) {
+  if (!config.has_value()) {
+    return nullptr;
+  }
+  if (!config->use_http_header()) {
+    return nullptr;
+  }
+  return std::make_unique<OriginalDstCluster::LoadBalancer::OriginalHostProvider>(
+      stats, config->has_use_http_authority() ? Http::Headers::get().Host
```

Comment:
`config->has_use_http_authority()` only checks the field is specified or not. But it is bool, it still has chance to get a false value. So we should check the value here.

Reply:
Good point! Need to rework